android/java/org/chromium/chrome/browser/playlist/hls_content/HlsUtils.java

```diff
@@ -92,6 +98,7 @@ public void onDataCompleted() {
                         };
         sPlaylistStreamingObserverImpl =
                 new PlaylistStreamingObserverImpl(playlistStreamingObserverImplDelegate);
+        playlistService.clearObserverForStreaming();
```

Comment:
what other observers are there? Why are we always clearing all of them?

Reply:
we are clearing explicitly because  we use the same call to download sequential files in the same service to keep a track on it. basically we get the main manifest file from m3u8 file, then we read the file to get different segments, we download different segments sequentially to make  sure the correct data.

Reply:
and all of that is happening in the same service as we need the content in the same final file to use it for playing.